File: src/main/java/com/primogemstudio/advancedui/render/filter/FastGaussianBlurFilter.java
package com.primogemstudio.advancedui.render.filter;

import com.mojang.blaze3d.pipeline.RenderTarget;
import com.mojang.blaze3d.pipeline.TextureTarget;

import java.util.Map;

import static com.primogemstudio.advancedui.render.Shaders.FAST_GAUSSIAN_BLUR;
import static net.minecraft.client.Minecraft.ON_OSX;

public class FastGaussianBlurFilter implements Filter {
    private static final TextureTarget target = new TextureTarget(1, 1, true, ON_OSX);
    private boolean enable;
    static {
        target.setClearColor(0, 0, 0, 0);
    }

    @Override
    public RenderTarget getTarget() {
        return target;
    }

    public void setArgs(Map<String, Object> data) {

    }

    @Override
    public void render(float partialTicks) {
        if (enable) {
            FAST_GAUSSIAN_BLUR.setSamplerUniform("InputSampler", target);
            FAST_GAUSSIAN_BLUR.render(partialTicks);
        }
    }

    @Override
    public void enable() {
        enable = true;
    }

    @Override
    public void reset() {
        enable = false;
    }
}

File: src/main/java/com/primogemstudio/advancedfmk/render/filter/FastGaussianBlurFilter.kt
package com.primogemstudio.advancedfmk.render.filter

import com.mojang.blaze3d.pipeline.RenderTarget
import com.mojang.blaze3d.pipeline.TextureTarget
import com.primogemstudio.advancedfmk.render.Shaders.FAST_GAUSSIAN_BLUR
import net.minecraft.client.Minecraft

class FastGaussianBlurFilter : Filter {
    private var enable = false
    override val target: RenderTarget
        get() = Companion.target

    override fun setArg(name: String?, value: Any?) {}
    override fun render(partialTicks: Float) {
        if (!enable) return
        FAST_GAUSSIAN_BLUR.setSamplerUniform("InputSampler", Companion.target)
        FAST_GAUSSIAN_BLUR.render(partialTicks)
    }

    override fun enable() {
        enable = true
    }

    override fun reset() {
        enable = false
    }

    companion object {
        private val target = TextureTarget(1, 1, true, Minecraft.ON_OSX)

        init {
            target.setClearColor(0f, 0f, 0f, 0f)
        }
    }
}
